aten/src/ATen/native/Convolution.cpp

```diff
@@ -1485,4 +1490,253 @@ std::tuple<Tensor,Tensor,Tensor> _convolution_double_backward( const c10::option
+  auto weight = weight_;
+  c10::MaybeOwned<Tensor> bias_maybe_owned = at::borrow_from_optional_tensor(bias_opt);
+  auto bias = *bias_maybe_owned;
+
+  auto k = weight.ndimension();
+  int64_t dim = k - 2;
+
+  TORCH_CHECK(dim > 0, "weight should have at least three dimensions");
+
+  auto& ctx = at::globalContext();
+  ConvParams params;
+  params.stride = expand_param_if_needed(stride, "stride", dim);
+  params.padding = expand_param_if_needed(padding, "padding", dim);
+  params.dilation = expand_param_if_needed(dilation, "dilation", dim);
+  params.transposed = transposed;
+  params.output_padding = expand_param_if_needed(output_padding, "output_padding", dim);
+  params.groups = groups;
+  params.benchmark = ctx.benchmarkCuDNN();
+  params.deterministic = ctx.deterministicCuDNN() || ctx.deterministicAlgorithms();
+  params.cudnn_enabled = ctx.userEnabledCuDNN();
+  params.allow_tf32 = ctx.allowTF32CuDNN();
+
+  // output_padding is only supported for transposed convolutions
+  if (!params.transposed) {
+    for (auto pad : params.output_padding) {
+      TORCH_CHECK(pad == 0, "output_padding is not supported for non-transposed convolutions; got: ",
+        params.output_padding);
+    }
+  }
+
+  if (k == 3) {
+    // avoid accidentally going through NHWC for permuted 3d input.
+    if (!input.is_mkldnn()) {
+      input = input.contiguous();
+    }
+    params.view1d_as_2d();
+    grad_output = view4d(grad_output);
+    input = view4d(input);
+    weight = view4d(weight);
+  }
+
+  // Select appropriate backend to use.
+  ConvBackend backend = select_conv_backend(input, weight, bias, params);
+  at::MemoryFormat backend_memory_format = determine_backend_memory_format(input, weight);
+
+  // Call the backend.
+  Tensor backend_grad_input, backend_grad_weight, backend_grad_bias;
+  auto kernel_size = weight.sizes().slice(2);
+  switch(backend) {
+    case ConvBackend::CudaDepthwise2d:
+      std::tie(backend_grad_input, backend_grad_weight) =
+        at::_conv_depthwise2d_backward(grad_output, input.contiguous(), weight, kernel_size,
+            params.stride, params.padding, params.dilation, {output_mask[0], output_mask[1]});
+      break;
+    case ConvBackend::CudaDepthwise3d:
+      TORCH_CHECK(input.ndimension() == 5);
+      std::tie(backend_grad_input, backend_grad_weight, backend_grad_bias) =
+        at::conv_depthwise3d_backward(
+          grad_output, input.contiguous(), weight, kernel_size, params.stride, params.padding, params.dilation,
+          output_mask);
+      break;
+    case ConvBackend::Cudnn:
+      check_input_same_type_as_parameters(input, weight, bias);
+      std::tie(backend_grad_input, backend_grad_weight) = at::cudnn_convolution_backward(
+          input.contiguous(backend_memory_format), grad_output, weight, params.padding, params.stride,
+          params.dilation, params.groups, params.benchmark, params.deterministic, params.allow_tf32,
+          {output_mask[0], output_mask[1]});
```

Comment:
Okay I'm probably missing something, for a lot of the entries the .contiguous() is in a different place in derivatives.yaml vs in this code here. 

![image](https://user-images.githubusercontent.com/5652049/142701872-a0f78028-6f2b-4b5f-af23-5a9d77d2bfef.png)